File: Animation/ViewCollapseDemo/app/src/main/java/com/tapzo/yashsoni/viewcollapsedemo/MainActivity.java
package com.tapzo.yashsoni.viewcollapsedemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    LinearLayout collapsableViewInfo;
    LinearLayout collapsableView;
    TextView collapsableViewText;
    ImageView collapsableViewImage;
    ExpandCollapseAnimator expandCollapseAnimator;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        collapsableView = (LinearLayout) findViewById(R.id.collapsable_view);
        collapsableViewInfo = (LinearLayout) findViewById(R.id.collapsable_view_info);

        collapsableViewText = (TextView) findViewById(R.id.collapsable_view_text);
        collapsableViewImage = (ImageView) findViewById(R.id.collapsable_view_image);

        expandCollapseAnimator = new ExpandCollapseAnimator(collapsableView, collapsableViewText);

        collapsableViewInfo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                expandCollapseAnimator.toggleView(collapsableView.getHeight());
                collapsableViewImage.setRotation(collapsableViewImage.getRotation() + 180);
            }
        });
    }
}

File: Kotlin/MyFirstKotlinApp/app/src/main/java/com/example/yashsoni/myfirstkotlinapp/MainActivity.kt
package com.example.yashsoni.myfirstkotlinapp

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_click_me.setOnClickListener({
            if (et_user_name.text.toString().isBlank() || et_user_age.text.toString().isBlank()) {
                Toast.makeText(this, "Please fill both fields", Toast.LENGTH_SHORT).show()
            } else {
                val userObj = MyData(et_user_name.text.toString(),
                        Integer.valueOf(et_user_age.text.toString()))
                val i = Intent(this, SecondActivity().javaClass)
//                i.putExtra("name", userObj.name)
//                i.putExtra("age", userObj.age)
                i.putExtra("user", userObj)
                startActivity(i)
            }
        })
    }
}
